### PR TITLE
Bottom bar screens loading state

### PR DESCRIPTION
**Pull request by robert-nganga**
# Scope
Added loading states for the sessions, feed, and home screens

_Please make sure to read the [Contribution Guidelines](https://github.com/droidconKE/droidconKE2023Android/blob/main/CONTRIBUTING.md)
and check that you understand and have followed it as best as possible Explain what your feature
does in a short paragraph._ please check the below boxes
- [x] I have followed the coding conventions
- [ ] I have added/updated necessary tests
- [x] I have tested the changes added on a physical device
- [ ] I have run `./codeAnalysis.sh` on linux/unix or `codeAnalysys.bat` on windows to make sure all lint/formatting checks have been done.

## Closes/Fixes Issues
_Declare any issues by typing `fixes #1` or `closes #1` for example so that the automation can kick
in when this is merged_

## Other testing QA Notes
_What have you tested specifically and what possible impacts/areas there are that may need retesting
by others._

Please add a screenshot (if necessary)

photo_2023-09-10_13-52-21
photo_2023-09-10_13-52-30
photo_2023-09-10_13-52-39

### CODE REVIEW

**Comment by wangerekaharun** [comment]
@robert-nganga great work on this, I just looked at the screenshots, and it looks good. One more request, can you do the same work for FeeScreen? Bo that we can do a full review

**Comment by robert-nganga** [comment]
> @robert-nganga great work on this, I just looked at the screenshots, and it looks good. One more request, can you do the same work for FeeScreen? Bo that we can do a full review

Which FeeScreen are you referring to?

**Comment by wangerekaharun** [comment]
> > @robert-nganga great work on this, I just looked at the screenshots, and it looks good. One more request, can you do the same work for FeeScreen? Bo that we can do a full review
> 
> Which FeeScreen are you referring to?

Sorry meant the FeedScreen

File: presentation/src/main/java/com/android254/presentation/feed/view/FeedScreen.kt [hunk 1]
```diff
@@ -120,7 +120,10 @@ fun FeedScreen(
                         verticalArrangement = Arrangement.Center,
                         horizontalAlignment = Alignment.CenterHorizontally
                     ) {
-                        CircularProgressIndicator()
+                        //CircularProgressIndicator()
```
Comment (chepsi) [inline presentation/src/main/java/com/android254/presentation/feed/view/FeedScreen.kt:123]: unnecessary comment?

**Comment by robert-nganga** [comment]
> > > @robert-nganga great work on this, I just looked at the screenshots, and it looks good. One more request, can you do the same work for FeeScreen? Bo that we can do a full review
> > 
> > 
> > Which FeeScreen are you referring to?
> 
> Sorry meant the FeedScreen

I've already done the FeedScreen. This PR includes the loading states for the HomeScreen, FeedScreen, and SessionsScreen. Am currently working on the AboutScreen.

File: presentation/src/main/java/com/android254/presentation/home/screen/HomeScreen.kt [hunk 2]
```diff
@@ -88,21 +90,35 @@ fun HomeScreen(
...
                 HomeSpacer()
-                if (homeViewState.isSessionsSectionVisible) {
-                    HomeSessionSection(
-                        sessions = homeViewState.sessions,
-                        onSessionClick = onSessionClicked,
-                        onViewAllSessionClicked = navigateToSessionScreen
-                    )
-                    HomeSpacer()
+                when{
+                    isSyncing && homeViewState.sessions.isEmpty() ->{
```
Comment (chepsi) [inline presentation/src/main/java/com/android254/presentation/home/screen/HomeScreen.kt:94]: is this necessary` homeViewState.sessions.isEmpty()`?

File: presentation/src/main/java/com/android254/presentation/sessions/components/SessionStateComponent.kt [hunk 3]
```diff
@@ -61,7 +61,8 @@ fun SessionsStateComponent(
     val swipeRefreshState = rememberSwipeRefreshState(isRefreshing = isRefreshing)
     when (sessionsUiState) {
         is SessionsUiState.Loading -> {
-            SessionsLoadingSkeleton()
+            //SessionsLoadingSkeleton()
```
Comment (chepsi) [inline presentation/src/main/java/com/android254/presentation/sessions/components/SessionStateComponent.kt:64]: Unnecessary comment

**Review by chepsi — APPROVED** [review]
This is amazing. 🥇

**Comment by robert-nganga** [comment]
I've updated the FeedScreen and removed the unnecessary comments.

**Pull request merged**